File: packages/silo-client/src/entries/entry-list-query.ts
import type { FilterExpression } from "../query/filter-expression.js";
import type { SortTerm } from "../query/sort-term.js";

/** Everything `collection.list()` accepts, all optional. `where`
 * builds through `collection.filter`; `sort` takes a built `SortTerm` or a
 * raw string. */
export interface EntryListQuery {
  where?: FilterExpression;
  sort?: SortTerm | string;
  limit?: number;
  offset?: number;
}

File: packages/silo-client-dart/lib/src/entries/entry_list_query.dart
import '../query/filter_expression.dart';

/// Everything `list` accepts. [sort] takes a `SortTerm` or a raw string.
final class EntryListQuery {
  const EntryListQuery({this.where, this.sort, this.limit, this.offset});

  final FilterExpression? where;
  final String? sort;
  final int? limit;
  final int? offset;

  EntryListQuery copyWith({int? limit, int? offset}) =>
      EntryListQuery(where: where, sort: sort, limit: limit ?? this.limit, offset: offset ?? this.offset);
}
